src/Microsoft.AspNetCore.Mvc.ViewFeatures/RemoteAttributeBase.cs

```diff
@@ -0,0 +1,214 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Globalization;
+using System.Linq;
+using Microsoft.AspNetCore.Mvc.DataAnnotations;
+using Microsoft.AspNetCore.Mvc.ModelBinding.Validation;
+using Microsoft.AspNetCore.Mvc.ViewFeatures;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Localization;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.Mvc
+{
+    /// <summary>
+    /// A <see cref="ValidationAttribute"/> which configures Unobtrusive validation to send an Ajax request to the
+    /// web site. The invoked endpoint should return JSON indicating whether the value is valid.
+    /// </summary>
+    /// <remarks>Does no server-side validation of the final form submission.</remarks>
+    [AttributeUsage(AttributeTargets.Property, AllowMultiple = false, Inherited = true)]
+    public abstract class RemoteAttributeBase : ValidationAttribute, IClientModelValidator
```

Comment:
What do you think about this base attribute @pranavkm  and @dougbu? I will commit derived attributes with unit tests and functional tests once you are OK with the base.

Reply:
This looks good to me.